### PR TITLE
New simpler configuration api

### PR DESCRIPTION
**Pull request by tsvetelintsonev**
Usage examples:

``` csharp
var logger = new LoggerConfiguration()
        .Enrich.FromLogContext()
        .WriteTo.RabbitMQ((clientConfiguration, sinkConfiguration) => {
            clientConfiguration.Username     = _config["RABBITMQ_USER"];
            clientConfiguration.Password     = _config["RABBITMQ_PASSWORD"];
            clientConfiguration.Exchange     = _config["RABBITMQ_EXCHANGE"];
            clientConfiguration.ExchangeType = _config["RABBITMQ_EXCHANGE_TYPE"];
            clientConfiguration.DeliveryMode = RabbitMQDeliveryMode.Durable;
            clientConfiguration.RouteKey     = "Logs";
            clientConfiguration.Port         = 5672;
            
            foreach (string hostname in _config["RABBITMQ_HOSTNAMES"]) {
                clientConfiguration.Hostnames.Add(hostname);
            }

            sinkConfiguration.TextFormatter  = new JsonFormatter();
        }).CreateLogger();
```

``` csharp
var config = new RabbitMQClientConfiguration
    {
        Port            = 5672,
        DeliveryMode    = RabbitMQ.RabbitMQDeliveryMode.Durable,
        Exchange        = "test_exchange",
        Username        = "guest",
        Password        = "guest",
        ExchangeType    = "fanout"
    };

foreach (string hostname in _config["RABBITMQ_HOSTNAMES"]) {
    config .Hostnames.Add(hostname);
}

var logger = new LoggerConfiguration()
    .WriteTo.RabbitMQ((clientConfiguration, sinkConfiguration) => {
    clientConfiguration.From(config);
    sinkConfiguration.TextFormatter = new JsonFormatter();
}) .CreateLogger();
```

``` csharp
var logger = new LoggerConfiguration()
    .WriteTo.RabbitMQ((clientConfiguration, sinkConfiguration) => {
        clientConfiguration.From(Configuration.Bind("RabbitMQClientConfiguration", new RabbitMQClientConfiguration()));
        sinkConfiguration.TextFormatter = new JsonFormatter();
}).CreateLogger();
```

### CODE REVIEW

File: src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/RabbitMQClientConfiguration.cs [hunk 1]
```diff
@@ -20,10 +20,9 @@ namespace Serilog.Sinks.RabbitMQ.Sinks.RabbitMQ
     /// <summary>
     /// Configuration class for RabbitMqClient
     /// </summary>
-    public class RabbitMQConfiguration
+    public class RabbitMQClientConfiguration
     {
-        public string Hostname { get; set; } = string.Empty;
-        public IList<string> Hostnames { get; } = new List<string>();
+        public IList<string> Hostnames { get; set; } = new List<string>();
```
Comment (huan086) [inline src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/RabbitMQClientConfiguration.cs]: I've tested that both `IConfiguration.GetSection("").Get<RabbitMQClientConfiguration>()` and  `IConfiguration.GetSection("").Bind(config)` works with readonly collection.

This line would thus be `public IList<string> Hostnames { get; } = new List<string>();`

Comment (tsvetelintsonev) [reply src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/RabbitMQClientConfiguration.cs]: Thanks for testing it out. I will update the Hostnames property.

**Comment by tsvetelintsonev** [comment]
All changes have been applied according to the comments.

**Comment by sonicjolt** [comment]
Could you update the instructions for this (Hostnames), .. I'm writing a test-host application for a docker-compose setup, and instructions are unclear how to populate Hostnames ;-)

**Comment by tsvetelintsonev** [comment]
I just updated the usage examples. Please say if there is something else you need to know regarding the api usage :-)

**Comment by sonicjolt** [comment]
With multiple hosts, is it unrealistic that the port-numbers will change ?

**Comment by tsvetelintsonev** [comment]
I will look into the docs whether it is possible to register multiple Hostnames with different ports. This makes me thinking about different messaging protocols per Hostname. Should we look at this as well?

**Comment by sonicjolt** [comment]
Let's wait for the feature request to come in instead, and wrap this milestone up

**Comment by tsvetelintsonev** [comment]
Should we wrap this PR?

**Comment by sonicjolt** [comment]
I think we should just keep it like this -- if it arises as a feature request, it will come in fairly quickly as a request.. I'll merge this PR.

**Pull request merged**